### PR TITLE
[Feat4-FR1] Revise Customer model and associated unit tests

### PR DESCRIPTION
**Pull request by bradyb2005**
## Description  
*Revised Customer model and associated tests.*

## Changes
- [x] Bug fix

## Checklist
- [x] All tests pass
- [x] Test documentation submitted to folder
- [x] Added necessary comments
- [x] Formatted correctly

#49 

image

### CODE REVIEW

File: backend/tests/customer/unit_tests/test_customer_model.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        customer = Customer(**test_customer_data)
+        
+        # Assert Basic Attributes
+        assert customer.id == 1
+        assert customer.username == "bbracken"
+        assert customer.phone == "250-222-2222"
+        
+        # Assert Mock Behavior
+        # Check that Cart was initialized with the right arguments
+        MockCart.assert_called_once_with(customer.id, customer)
```
Comment (Graysonzz) [inline backend/tests/customer/unit_tests/test_customer_model.py]: You have hardcoded customer id as 1 but are now using customer.id. Please use your mocking fixture and keep it hardcoded as 1. If customer.id breaks, it might still pass tests so this ensures the proper data is passed

File: backend/tests/customer/unit_tests/test_customer_model.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# python3 -m pytest backend/tests/customer/unit_tests/test_customer_model.py
+import pytest
+from unittest.mock import patch
+from backend.models.user.customer_model import Customer
+from backend.models.cart.cart_model import Cart
```
Comment (Graysonzz) [inline backend/tests/customer/unit_tests/test_customer_model.py]: Cart model does not currently exist in this branch or main. This means that no tests can properly run meaning it cannot be merged. Please fix

**Review by aratnala7 — CHANGES_REQUESTED** [review]
I agree with Grayson’s review. The main concerns raised about the missing Cart model, the possible over-coupling from passing self into Cart, the test assertion consistency, and the missing testing explanation should be addressed before merge.

One additional small notewoyld be submit_review( ) is still a placeholder with pass, so it may be worth clarifying in the PR description or comments that this is intentional and not part of the implemented functionality for this change,

File: backend/models/user/customer_model.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        self.items = []
+
+@dataclass
+class Customer(User):
+    phone: str
+    address: str
+    city: str
+    postal_code: str
+
+    """
```
Comment (Graysonzz) [inline backend/models/user/customer_model.py]: These comments are unnecessary since the file does it! Can be deleted to improve format consistency and reduce redundancy

File: backend/tests/customer/unit_tests/test_customer_model.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        "password_hash": "supersecret123",
+        "phone": "250-222-2222",
+        "address": "1234 University Way",
+        "city": "Kelowna",
+        "postal_code": "V1V 1V1"
+    }
+
+## --- Initialization Tests ---
+
+def test_customer_init(test_customer_data):
```
Comment (Graysonzz) [inline backend/tests/customer/unit_tests/test_customer_model.py]: I dont think you have to check that every single attribute is set correctly since you used parametrize. If you choose one and its good, you can assume the other are too since they'd be passed in the same call. Also maybe add a check that customer.cart.customer does equal customer just to check circular linking

**Review by Graysonzz — CHANGES_REQUESTED** [review]
This looks great Brady! Overall your logic is sound and I think the revisions are great! I had a few suggestions to reduce redundancy in your code because right now your code is inefficient and violates DRY (Don't Repeat Yourself). I've also recommended adding one more edge case! I will test once these changes have been made!
